core/lib/zksync_core/src/api_server/web3/namespaces/eth.rs

```diff
@@ -826,6 +826,10 @@ impl<G: L1GasPriceProvider> EthNamespace<G> {
                     .state
                     .resolve_filter_block_number(filter.to_block)
                     .await?;
+                let latest_block = self
+                    .state
+                    .resolve_filter_block_number(Some(BlockNumber::Latest))
+                    .await?;
```

Comment:
Why is this constantly `BlockNumber::Latest` ? I believe that this at least should look to what user has specified in `to_block` and if that is `None` default to `Latest` or whatever the ethereum spec says?

Reply:
Generally I think that it shouldn't be a problem, since we use it only when updating the `from_block` when we are actually comparing what is the current `to_block` and this one